From 340b200c6f0efe8dbe38186a5c94ba28cd9bfd5f Mon Sep 17 00:00:00 2001
From: Davis Goodin <dagood@microsoft.com>
Date: Mon, 21 Dec 2020 15:17:22 -0600
Subject: [PATCH] Don't use RTMVersion in source-build

This seems like it would bring in prebuilts, and more directly, causes:

  Detected package downgrade: System.Security.Cryptography.Pkcs from 5.0.1 to 5.0.0. Reference the package directly from the project to select a different version. 
   RepoTasks -> NuGet.Packaging 5.8.0-rc.6930 -> System.Security.Cryptography.Pkcs (>= 5.0.1) 
   RepoTasks -> System.Security.Cryptography.Pkcs (>= 5.0.0)
---
 eng/Dependencies.props | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/eng/Dependencies.props b/eng/Dependencies.props
index d573a7bde9..79436b5a16 100644
--- a/eng/Dependencies.props
+++ b/eng/Dependencies.props
@@ -198,7 +198,11 @@ and are generated based on the last package release.
     <!-- Get versions. -->
     <LatestPackageReference Update="@(LatestPackageReference)">
       <Version>$(%(VersionName)PackageVersion)</Version>
-      <RTMVersion>$(%(VersionName)V0PackageVersion)</RTMVersion>
+      <!--
+        Don't set RTMVersion if building from source. RTMVersion represents a downlevel version of
+        the package, which we don't have access to in source-build. (It would be a prebuilt.)
+      -->
+      <RTMVersion Condition="'$(DotNetBuildFromSource)' != 'true'">$(%(VersionName)V0PackageVersion)</RTMVersion>
 
       <!-- Remove excess metadata. -->
       <RTMVersion Condition=" '%(Version)' == '%(RTMVersion)' " />
-- 
2.25.4

